### PR TITLE
perf: reduce allocations and calls to ToArray

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: src/Adapter/MSTestAdapter.PlatformServices/Utilities/DeploymentUtilityBase.cs [hunk 1]
```diff
@@ -272,7 +272,7 @@ protected IEnumerable<string> Deploy(IList<DeploymentItem> deploymentItems, stri
     /// <param name="isDirectory">Is this a directory.</param>
     /// <returns>Paths to items to deploy.</returns>
     [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Requirement is to handle all kinds of user exceptions and message appropriately.")]
-    protected string[]? GetFullPathToFilesCorrespondingToDeploymentItem(DeploymentItem deploymentItem, string testSource, string resultsDirectory, IList<string> warnings, out bool isDirectory)
+    protected List<string>? GetFullPathToFilesCorrespondingToDeploymentItem(DeploymentItem deploymentItem, string testSource, string resultsDirectory, IList<string> warnings, out bool isDirectory)
```
Comment (MarcoRossignoli) [inline src/Adapter/MSTestAdapter.PlatformServices/Utilities/DeploymentUtilityBase.cs:275]: IList/IReadOnlyList?

File: src/Adapter/MSTestAdapter.PlatformServices/Utilities/ReflectionUtility.cs [hunk 2]
```diff
@@ -129,24 +125,26 @@ internal class ReflectionUtility
...
-        return type == null ? memberInfo.GetCustomAttributes(inherit).ToArray() : memberInfo.GetCustomAttributes(type, inherit).ToArray();
+        return type == null
+            ? memberInfo.GetCustomAttributes(inherit)
+            : memberInfo.GetCustomAttributes(type, inherit);
 #endif
     }
 
 #if NETFRAMEWORK
-    internal static object[] GetCustomAttributes(Assembly assembly, Type type)
+    internal static List<Attribute> GetCustomAttributes(Assembly assembly, Type type)
```
Comment (MarcoRossignoli) [inline src/Adapter/MSTestAdapter.PlatformServices/Utilities/ReflectionUtility.cs:138]: ditto

**Review by MarcoRossignoli — APPROVED** [review]
A profiling before/after could be interesting

File: src/Adapter/MSTestAdapter.PlatformServices/Utilities/DeploymentUtilityBase.cs (hunk 1 above)
Comment (Evangelink) [reply src/Adapter/MSTestAdapter.PlatformServices/Utilities/DeploymentUtilityBase.cs:275]: This is only internal/private stuff so I prefer to always be as specific as possible.

**Pull request merged**